src/haddock/libs/libplots.py

```diff
@@ -523,29 +561,317 @@ def scatter_plot_handler(capri_filename, cl_rank, png, dpi):
+    number_of_cols = SCATTER_MATRIX_SIZE[1]
+    number_of_rows = int(np.ceil(number_of_plots / number_of_cols))
+    # enable horizontal scroll
+    width = 600 if number_of_clusters > 5 else 350
+    height = 600 if number_of_clusters > 5 else 350
+    return number_of_rows, number_of_cols, width, height
+
+
+def report_plots_handler(
+        plots,
+        plot_title,
+        shared_xaxes=False,
+        shared_yaxes=False
+        ):
+    """
+    Create a figure that holds subplots.
+
+    The idea is that for each type (scatters or boxes), the individual plots are
+    considered subplots. In the report, some of the axes are shared. The
+    settings for sharing axes depends on the type (scatters or boxes).
+
+    Parameters
+    ----------
+    plots : list
+        list of plots generated by `analyse` command
+    plot_title: str
+        title of the plot.
+    shared_xaxes: boolean or str (default False)
+        a paramtere of plotly.subplots.make_subplots
+    shared_yaxes: boolean or str (default False)
+        a paramtere of plotly.subplots.make_subplots
+
+    Returns
+    -------
+    fig :
+        an instance of plotly.graph_objects.Figure
+    """
+    number_of_rows, number_of_cols, width, height = _report_grid_size(plots)
+    fig = make_subplots(
+        rows=number_of_rows,
+        cols=number_of_cols,
+        shared_xaxes=shared_xaxes,
+        shared_yaxes=shared_yaxes,
+        vertical_spacing=(0.4 / number_of_rows),
+        horizontal_spacing=(0.3 / number_of_cols),
+        )
+    for i, sub_fig in enumerate(plots):
+        col_index = int((i % number_of_cols) + 1)
+        row_index = int(np.floor(i / number_of_cols) + 1)
+        # hide legend of plots except one
+        if i != 0:
+            sub_fig.for_each_trace(lambda trace: trace.update(showlegend=False))
+        fig.add_traces(sub_fig.data, rows=row_index, cols=col_index)
+        fig.update_yaxes(
+            title_text=sub_fig.layout.yaxis.title.text,
+            row=row_index,
+            col=col_index,
+            title_standoff=5,
+            automargin=True,
+            )
+        # x title only on the last row
+        if shared_xaxes == "all":
+            row_index = number_of_rows
+        fig.update_xaxes(
+            title_text=sub_fig.layout.xaxis.title.text,
+            row=row_index,
+            col=col_index,
+            title_standoff=5,
+            automargin=True,
+            )
+        legend_title_text = sub_fig.layout.legend.title.text
+    fig.update_layout(
+        title_text=plot_title,
+        legend_title_text=legend_title_text,
+        height=height * number_of_rows,
+        width=width * number_of_cols,
+        )
+    return fig
+
+
+def find_best_struct(ss_file, number_of_struct=4):
+    """
+    Find best structures.
+
+    It inspects model-cluster-ranking recorded in capri_ss.tsv file and finds
+    the best models (models with lower ranks). By default, it selects the 4 best
+    models.
```

Comment:
I agree with @amjjbonvin here: the number of structures to choose depend on many parameters, and the data we show here can disagree with those reported in previous steps of the workflow (for example the `clustfcc` step).

I see three potential solutions here:
1. we report mean and standard deviation over the full set of structures belonging to a cluster
2. we choose 4 as the default number (in this case we should state it clearly, e.g. "data refer to the first 4 structures")
3. (not so immediate) we make the analysis read this number from the `capri_clt.tsv` file, where the value of `clt_threshold` is reported